pytorch_transformers/tokenization_xlm.py

```diff
@@ -122,16 +201,16 @@ def __init__(self, vocab_file, merges_file, unk_token="<unk>", bos_token="<s>",
                                            cls_token=cls_token, mask_token=mask_token,
                                            additional_special_tokens=additional_special_tokens,
                                            **kwargs)
-        try:
-            import ftfy
-            from spacy.lang.en import English
-            _nlp = English()
-            self.nlp = _nlp.Defaults.create_tokenizer(_nlp)
-            self.fix_text = ftfy.fix_text
-        except ImportError:
-            logger.warning("ftfy or spacy is not installed using BERT BasicTokenizer instead of SpaCy & ftfy.")
-            self.nlp = BasicTokenizer(do_lower_case=True)
-            self.fix_text = None
+
+        # cache of sm.MosesPunctNormalizer instance
+        self.cache_moses_punct_normalizer = dict()
+        # cache of sm.MosesTokenizer instance
+        self.cache_moses_tokenizer = dict()
+        self.lang_with_custom_tokenizer = set(['zh', 'th', 'ja'])
+        # True for current supported model (v1.2.0), False for XLM-17 & 100
+        self.do_lowercase_and_remove_accent = True
```

Comment:
So here the workflow for the user is to manually deactivate this for XLM-17 & 100 (e.g. by doing `tokenizer. do_lowercase_and_remove_accent = False` after loading the tokenizer)?

This looks a bit error-prone to me. Maybe we should have a way to setup a configuration for the provided tokenizer. I was thinking about that for Bert already (people forget to set `do_lower_case` regularly).

I'll add something high level on the base class `PreTrainedTokenizer`.

Reply:
Sounds good to me indeed

Reply:
Ok added this